### DISCUSSION TITLE
How to uninstall Shield-oauth?

### DISCUSSION THREAD

**Discussion by darrenfauth** [datamweb/shield-oauth#14] [Q&A] [answered]
How to I disable this? I'm not sure if this is the source of an issue I now have with the debugbar or not.  At some point after installing shield-oauth I noticed I had a 500 Internal Server Error with displaying my  debugbar.
```
http://dgg2023.test/?debugbar_time=1667672584.295774
SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data

From my log file:
INFO - 2022-11-05 13:29:12 --> Session: Class initialized using 'CodeIgniter\Session\Handlers\FileHandler' driver.
INFO - 2022-11-05 13:29:12 --> Session: Class initialized using 'CodeIgniter\Session\Handlers\FileHandler' driver.
CRITICAL - 2022-11-05 13:29:12 --> foreach() argument must be of type array|object, string given
in VENDORPATH\datamweb\shield-oauth\src\Libraries\Basic\ShieldOAuth.php on line 86.
 1 VENDORPATH\datamweb\shield-oauth\src\Libraries\Basic\ShieldOAuth.php(86): CodeIgniter\Debug\Exceptions->errorHandler(2, 'foreach() argument must be of type array|object, string given', 'VENDORPATH\\datamweb\\shield-oauth\\src\\Libraries\\Basic\\ShieldOAuth.php', 86)
 2 VENDORPATH\datamweb\shield-oauth\src\Libraries\Basic\ShieldOAuth.php(111): Datamweb\ShieldOAuth\Libraries\Basic\ShieldOAuth->otherOAuth()
 3 VENDORPATH\datamweb\shield-oauth\src\Views\Decorators\ShieldOAuth.php(21): Datamweb\ShieldOAuth\Libraries\Basic\ShieldOAuth->makeOAuthButton('login')
 4 SYSTEMPATH\View\ViewDecoratorTrait.php(31): Datamweb\ShieldOAuth\Views\Decorators\ShieldOAuth::decorate('<table>
```

Thanks!

******* EDIT ********
I figured out how to do it. I think my issue is more understanding how Github works to update my local files with changes that you made to the repo.

Great package and I'll be back for it once I learn more about Github.

**Comment by datamweb** [comment, marked as answer]
Hi @darrenfauth ,
I'm sorry for the problem, this has already been fixed in the `develop` branch, please update with the command below.

```console
composer require datamweb/shield-oauth:dev-develop
```

**Reply by darrenfauth** [reply]
Thank you so much for the quick reply. I love your addition to Shield with Oauth! I appreciate your contributions brother. I saw you had updated the files recently - but yeah, I got overwhelmed with the git commands.

**Reply by datamweb** [reply]
Welcome, 
we've all been in your situation.
I understand you more than anyone else.

Your message reminded me of a similar situation I had a few months ago.
Check here to find out how I was like you.
https://github.com/codeigniter4/CodeIgniter4/pull/5928#issuecomment-1113883822

With the help of good people, I was encouraged to learn and now I have a little better situation.

The topics you mentioned are completely separate.

Composer is for managing dependencies, when I got to know it, it was one of the most confusing things that confused me, but as time passed, I realized what wonderful features I had missed for years.

GitHub provides a host to maintain codes and facilities to participate in the project and...

Git allows developers or contributors to have versioning facilities, Git is extremely useful.
Don't be in a hurry to learn the above, try to get to know them with interest and patience.

Let me know anytime you need help
